extension/cgroupruntimeextension/integration_test.go

```diff
@@ -220,6 +257,13 @@ func TestCgroupV2SudoIntegration(t *testing.T) {
 			})
 			require.NoError(t, err)
 
+			if test.setECSMetadataURI {
+				server := startMockECSServer()
+				defer server.Close()
```

Comment:
Wdyt if we move this defer calls to the [t.Cleanup function](https://pkg.go.dev/testing#T.Cleanup)? That way is aligned with the other resources rollback strategy, and we are also safe on panics.

Reply:
I am unsure how to implement it as I see multiple options. I need to clean only if `test.setECSMetadataURI` is `true`. Because I find it handy to start the mock http server only if this same bool is true.

Reply:
Makes sense to me, I think you could append the cleanup function inside the conditional. Something like:

```go
			if test.setECSMetadataURI {
				server := startMockECSServer()
				os.Setenv("ECS_CONTAINER_METADATA_URI_V4", server.URL)
				t.Cleanup(func() {
					server.Close()
					os.Unsetenv("ECS_CONTAINER_METADATA_URI_V4")
				})
			}
```

Reply:
Should be resolved with latest commit https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36920/commits/6212bdef7a0c2476b682e1fe3e18e5d13e088f04